src/System.Text.JsonLab/System/Text/Json/Utf8Json.Reader.cs

```diff
@@ -422,7 +422,7 @@ private void EndObject()
                 }
                 else
                 {
-                    _utf8Json._state._inObject = _utf8Json._state._stack.Pop() != JsonTokenType.StartArray;
+                    _utf8Json._state._inObject = (JsonTokenType)_utf8Json._state._stack.Pop() != JsonTokenType.StartArray;
```

Comment:
Should we cache state in a local? i.e. would it improve perf?

Reply:
It might, and I can try it.